builtins/amp-video.js

```diff
@@ -101,6 +124,80 @@ export function installVideo(win) {
     isVideoSupported_() {
       return !!this.video_.play;
     }
+
+    /**
+     * Safari on iOS does not trigger the `canplay` event for muted inline
+     * videos despite the fact that calls to `play` are allowed for muted inline
+     * videos in iOS 10.
+     * It does trigger `canplay` if `autoplay` attribute is set however.
+     * Since VideoManager handles autoplay behaviour for AMP, we do not
+     * propagate the `autoplay` attribute to the underlying video element but
+     * to get around the iOS bug, we will temporarily propagate `autoplay`
+     * attribute until we get the `canplay` event and then remove it.
+     * {@see https://bugs.webkit.org/show_bug.cgi?id=161804}
+     * @private
+     */
+    fixIOSCanplayEventForAutoplay_() {
+      if (!this.element.hasAttribute('autoplay') || !this.platform_.isIos()) {
+        return;
+      }
+
+      this.propagateAttributes(['autoplay'], this.video_);
+      listenOnce(this.video_, VideoEvents.CANPLAY, () => {
```

Comment:
sorry I didn't see your replies until now. 
This a purely a hack that's is required for iOS. 
iOS (wrongly) does not fire 'canplay' event for muted inline videos (I have filed a bug with them). We rely on that event. iOS however does fire that event if `autoplay` attribute is set. Note that we do not set `autoplay` attribute on video elements since we manage autoplay behaviour ourselves (reason being browser's UX implementation of when to start playing/pausing is not inline with that we want). So what this  hack does is 1- sets `autoplay` to ensure iOS fires the 'canplay` event, then 2- removes the `autoplay` attribute after we got what we wanted.